src/ParallelAlgorithms/Interpolation/Callbacks/SendGhWorldtubeData.hpp

```diff
@@ -25,12 +27,13 @@ namespace callbacks {
 ///
 /// This is an InterpolationTargetTag::post_interpolation_callback;
 /// see InterpolationTarget for a description of InterpolationTargetTag.
-template <typename CceEvolutionComponent, typename TemporalId>
-struct SendGhWorldtubeData {
-  template <typename DbTags, typename Metavariables>
+template <typename CceEvolutionComponent>
+struct SendGhWorldtubeData
+    : tt::ConformsTo<intrp::protocols::PostInterpolationCallback> {
+  template <typename DbTags, typename Metavariables, typename TemporalId>
   static void apply(const db::DataBox<DbTags>& box,
                     Parallel::GlobalCache<Metavariables>& cache,
-                    const typename TemporalId::type& temporal_id) {
+                    const TemporalId& temporal_id) {
```

Comment:
If it expects a `TimeStepId` then it should just `static_assert(std:is_same_v<TemporalId, TimeStepId>)` in the `apply` function. I don't really like changing the nice general protocol to account for this one specific case. It shouldn't be the job of the protocol to account for specific use cases. Classes should conform to the protocol, and then if the class has any extra restrictions it should impose those itself.

Using this method of `static_assert` doesn't break the protocol because the protocol only looks at the function signature, and nothing about the internals. Does this seem like a good compromise?

Reply:
Ok, I'll have to think more if agree with your reasoning, but your suggestion seems like a reasonable solution for now 👍 Please add to the docs of `PostInterpolationCallback` that the apply function is expected to take any `TemporalId` type.